packages/@webex/plugin-cc/src/services/core/WebSocket/connection-service.ts

```diff
@@ -89,29 +82,24 @@ export class ConnectionService extends EventTarget {
     this.connectionProp = prop;
   }
 
-  private onPing = (event: Event): void => {
-    const msg = (event as CustomEvent<string>).detail;
-    const parsedEvent = JSON.parse(msg);
+  private onPing = (event: any): void => {
+    const parsedEvent = JSON.parse(event);
     if (this.reconnectingTimer) {
       clearTimeout(this.reconnectingTimer);
     }
     if (this.restoreTimer) {
       clearTimeout(this.restoreTimer);
     }
     this.isKeepAlive = parsedEvent.keepalive === 'true';
-    const shouldUpdateConnectionData =
-      this.isKeepAlive || (this.isConnectionLost && !this.isRestoreFailed);
-    const shouldDispatchEvent =
-      this.isKeepAlive || (this.isConnectionLost && !this.isRestoreFailed);
-    const shouldDispatchEventWithReconnect = this.isSocketReconnected && this.isKeepAlive;
 
-    if (shouldUpdateConnectionData) {
+    if (
```

Comment:
This old logic from agent desktop was working (my logic was not), hence re-added it here.